src/main/kotlin/com/cultureamp/eventsourcing/PGSessionAdvisoryBookmarkLock.kt

```diff
@@ -0,0 +1,77 @@
+package com.cultureamp.eventsourcing
+
+import org.slf4j.LoggerFactory
+import java.sql.Connection
+import java.util.UUID
+
+private val logger = LoggerFactory.getLogger(PGSessionAdvisoryBookmarkLock::class.java)
+class PGSessionAdvisoryBookmarkLock(
+    private val connectionProvider: () -> Connection,
+) : BookmarkLock {
+    private val lock = Any()
+    internal var connection = getConnection()
```

Comment:
actually same here, do we need to do anything to make this threadsafe since it's being access across thread / potentially reset by one thread which needs to be visible to the other threads